### PR TITLE
@craigspaeth - Rate limiter

### PR DESCRIPTION
**Pull request by broskoski**
cc @alloy @joeyAghion 

Mostly for discussion, with the values I set it would ratelimit anyone requesting 60 times in one minute (which now that I think about it is probably way too aggressive).

### CODE REVIEW

**Comment by craigspaeth** [comment]
👍  Works for me. I general, it'd probably be nice to have the option to configure on rate limiting.

That said, I do want to be aware of when something is actually reckless abusive behavior and deserves banning/limiting vs. if we should be addressing the resource consumption on our end instead.

File: lib/setup.coffee [hunk 1]
```diff
@@ -76,6 +86,8 @@ module.exports = (app) ->
   # Inject sharify data before anything
   app.use sharify
 
+  #
```
Comment (craigspaeth) [inline lib/setup.coffee]: Stray comment?

File: config.coffee [hunk 2]
```diff
@@ -13,6 +13,8 @@ module.exports =
   API_URL: 'http://localhost:3000'
   POSITRON_URL: 'http://localhost:3005'
   FUSION_URL: null
+  REQUEST_LIMIT: 60
+  REQUEST_EXPIRE_MS: 60000 # 1 minute
```
Comment (craigspaeth) [inline config.coffee]: This probably is way too aggressive, but 👍  for making it a config var that we can tune up and down at any point.

**Comment by joeyAghion** [comment]
I worry about these low-level enforcements that become difficult to work around later. 1 request per second seems reasonable, but I assume this is limiting by IP address? So all of Artsy HQ will potentially make requests as a single IP, and it's easy to imagine requests dropping frequently as a result. A worse situation would be an auction or other circumstance of spiky traffic where many devices may be on the same network.

Also, are there still asynchronous requests made from the Force front-end to its back-end (e.g. infinite scrolls)? Those could multiply the frequency of requests made by non-malicious visitors.

If we experiment with this, we should start with a generous threshold and ensure instances of thresholds being met are easily find-able in logs. We might want to _only_ log for a while, so we can eventually tune the middleware to only block under the most extenuating circumstances.

File: lib/setup.coffee [hunk 3]
```diff
@@ -53,6 +53,19 @@ ipfilter = require 'express-ipfilter'
...
+  # rate limited
+  limiter = require('express-limiter')(app, cache.client)
+  limiter
+    path: '*'
+    method: 'all'
+    lookup: ['headers.x-forwarded-for']
+    total: REQUEST_LIMIT
+    expire: REQUEST_EXPIRE_MS
+    onRateLimited: -> (req, res, next) 
+      console.log 'Rate limit exceeded for', req.headers['x-forwarded-for']
```
Comment (broskoski) [inline lib/setup.coffee]: @joeyAghion updated to _just_ log when a rate limit would happen.

**Comment by alloy** [comment]
Great feedback, @joeyAghion 👍

**Pull request merged**